src/main/webapp/portal/main/controllers.js

```diff
@@ -76,7 +76,23 @@ define(['angular','require'], function(angular, require) {
             var today = Date.parse(new Date());
             var startDate = Date.parse(new Date($scope.latestFeature.popup.startYear, $scope.latestFeature.popup.startMonth, $scope.latestFeature.popup.startDay));
             var endDate = Date.parse(new Date($scope.latestFeature.popup.endYear, $scope.latestFeature.popup.endMonth, $scope.latestFeature.popup.endDay));
-            if ((today > startDate && today < endDate) && $localStorage.lastSeenFeature < $scope.latestFeature.id) {
+            
+            // handle legacy local storage
+            if ($localStorage.lastSeenFeature === -1) {
+              if ($localStorage.hasSeenWelcome) {
+                $localStorage.lastSeenFeature = 1;
+              } else {
+                $localStorage.lastSeenFeature = 0;
+              }
```

Comment:
Only suggestion would be do a `delete $localStorage.hasSeenWelcome;` after the if/else block. #cleanup